### PR TITLE
Service Registration: History

### PR DESCRIPTION
**Pull request by kmh287**
This pull request distinguishes installing/registering the history service and creating it.Please see #7839 for more context.

/to @choumx

### CODE REVIEW

File: src/service/history-impl.js [hunk 1]
```diff
@@ -832,18 +836,24 @@ function createHistory(ampdoc) {
   } else {
     // Only one global "natural" binding is allowed since it works with the
     // global history stack.
-    binding = fromClass(ampdoc.win, 'global-history-binding',
-        HistoryBindingNatural_);
+    registerServiceBuilder(ampdoc.win,
+                           'global-history-binding',
+                           HistoryBindingNatural_);
```
Comment (choumx) [inline src/service/history-impl.js]: Nit: +4 spaces from start of previous line.

File: src/service/history-impl.js [hunk 2]
```diff
@@ -832,18 +836,24 @@ function createHistory(ampdoc) {
...
 export function installHistoryServiceForDoc(ampdoc) {
-  return /** @type {!History} */ (getServiceForDoc(ampdoc, 'history',
-      ampdoc => createHistory(ampdoc)));
+  registerServiceBuilderForDoc(
+      ampdoc,
+      'history',
+      /* opt_constructor */ undefined,
+      ampdoc => {
+        return createHistory(ampdoc);
+      });
```
Comment (choumx) [inline src/service/history-impl.js]: Nit: I like the short form arrow function.

File: test/functional/test-standard-actions.js [hunk 3]
```diff
@@ -150,7 +153,8 @@ describes.sandboxed('StandardActions', {}, () => {
 
   describe('"AMP" global target', () => {
     it('should implement goBack', () => {
-      const history = window.services.history.obj;
+      installHistoryServiceForDoc(ampdoc);
+      const history = historyForDoc(ampdoc);
```
Comment (choumx) [inline test/functional/test-standard-actions.js]: Thanks.

File: src/service/history-impl.js (hunk 1 above)
Comment (kmh287) [inline src/service/history-impl.js]: done

File: src/service/history-impl.js (hunk 2 above)
Comment (kmh287) [inline src/service/history-impl.js]: done

**Pull request merged**